WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift

```diff
@@ -89,8 +132,74 @@ public final class UserRepository: UserRepositoryProtocol {
         }
     }
 
+    public func updateUser(
+        from event: UserUpdateEvent
+    ) async throws {
+        try await context.perform { [self] in
+
+            let user = fetchOrCreateUser(
+                with: event.id
+            )
+
+            if isFederationEnabled {
+                user.domain = event.userID.domain
+            }
+
+            if let name = event.name {
+                user.name = name
+            }
+
+            if let email = event.email {
+                user.emailAddress = email
+            }
+
+            if let handle = event.handle {
+                user.handle = handle
+            }
+
+            if let accentColor = event.accentColorID {
+                user.accentColorValue = Int16(accentColor)
+            }
+
+            let assetKeys: Set<String> = [
+                ZMUser.previewProfileAssetIdentifierKey,
+                ZMUser.completeProfileAssetIdentifierKey
+            ]
+
+            /// Do not update assets if user has local modifications.
```

Comment:
Hmm... I wonder why 🤔

Reply:
Not sure why either, the documentation on `modifiedKeys` indicates:  `/// Keys that have been modified since the last save and not been synced and reset`. So, my assumption would be if user has local changes to its assets we don't want to update them and keep these changes until they're synced.

Reply:
Maybe worth adding that in the comment, even if it's just a guess.

Reply:
done in https://github.com/wireapp/wire-ios/pull/1958/commits/7457f6e25024c3c1f13930565644358273aa5112